src/app/core/services/schedule/schedule-generator.service.ts

```diff
@@ -218,6 +230,22 @@ export class ScheduleGeneratorService {
     return { schedule, completed }
   }
 
+  shiftDayOfWeek(refTimestamp, dayOfWeek) {
+    // NOTE: Shift ref timestamp to specified day of the same week
+    const moment = this.localization.moment(refTimestamp)
+    const target = moment.day(dayOfWeek).valueOf()
+    return moment.valueOf() <= target
+      ? moment
+          .day(dayOfWeek)
+          .toDate()
+          .getTime()
+      : moment
+          .add(1, 'w')
+          .day(dayOfWeek)
+          .toDate()
+          .getTime()
+  }
```

Comment:
Do we need to add 1 week? Should it not always start from the previous day of the week?